advertools/spider.py

```diff
@@ -949,9 +951,14 @@ def crawl(url_list, output_file, follow_links=False,
             else:
                 setting = '='.join([key, str(val)])
             settings_list.extend(['-s', setting])
+    if meta:
+        proxy = meta["proxy"]
+        rnd_proxy = random.choice(proxy)
```

Comment:
If a random proxy is chosen here, one proxy will be sent to the spider, and all requests will be using that proxy. 
The random choice has to be made every time a request is made, wether in list mode, or when following links.
I think the `meta` argument should be sent as is, saved to the spider class, and then the spider selects one randomly, right before making a request.

We can create a simple helper function maybe something like this (or something else), because this will be uses in 2-3 places:

```python
def _meta_param(meta):
    if meta.get("proxy"):
        meta['proxy'] = random.choice(meta['proxy'])
    return meta
```

This assumes that the proxy argument is a list of course. 

Then with each `Request` we can pass `meta=_meta_param(meta)`

What do you think?

Reply:
@eliasdabbas Please review latest commit. Thanks.